lib/wakes.rb

```diff
@@ -39,8 +39,12 @@ def self.create!(*args)
     build(*args).tap(&:save!)
   end
 
-  def self.redirect(source, target, label = nil)
-    RedirectMapper.new(source, target, label)
+  def self.redirect(source_path, target_path, label = nil)
+    RedirectMapper.new(source_path, target_path, label)
+  end
+
+  def self.redirect_with_hosts(source_host: nil, source_path:, target_host: nil, target_path:, label: nil)
```

Comment:
@benhutton I was going to ask you about this on Friday if you were still around (this was definitely a decision point).

I went this way for now because I wasn't sure we wanted to extend this quasi-URI syntax all the way to the API. (Right now it is just confined to internal implementation and specs.) I'm not opposed to taking it the rest of the way, though.